### PR TITLE
11957 parameterless constructor

### PR DESCRIPTION
**Pull request by nrkirby**
## Summary

The PR replaces all occurrences of Default constructor with Parameterless constructor and default constructor with parameterless constructor.

Fixes #Issue_Number #11957

### CODE REVIEW

File: docs/fsharp/language-reference/attributes.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ In the previous syntax, the *target* is optional and, if present, specifies the
...
 
-The *arguments* are the arguments to the constructor for the attribute type. If an attribute has a default constructor, the argument list and parentheses can be omitted. Attributes support both positional arguments and named arguments. *Positional arguments* are arguments that are used in the order in which they appear. Named arguments can be used if the attribute has public properties. You can set these by using the following syntax in the argument list.
+The *arguments* are the arguments to the constructor for the attribute type. If an attribute has a parameterless constructor, the argument list and parentheses can be omitted. Attributes support both positional arguments and named arguments. *Positional arguments* are arguments that are used in the order in which they appear. Named arguments can be used if the attribute has public properties. You can set these by using the following syntax in the argument list.
```
Comment (cartermp) [inline docs/fsharp/language-reference/attributes.md]: Please revert

File: docs/fsharp/language-reference/generics/constraints.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ There are several different constraints you can apply to limit the types that ca
 |Type Constraint|*type-parameter* :&gt; *type*|The provided type must be equal to or derived from the type specified, or, if the type is an interface, the provided type must implement the interface.|
 |Null Constraint|*type-parameter* : null|The provided type must support the null literal. This includes all .NET object types but not F# list, tuple, function, class, record, or union types.|
 |Explicit Member Constraint|[(]*type-parameter* [or ... or *type-parameter*)] : (*member-signature*)|At least one of the type arguments provided must have a member that has the specified signature; not intended for common use. Members must be either explicitly defined on the type or part of an implicit type extension to be valid targets for an Explicit Member Constraint.|
-|Constructor Constraint|*type-parameter* : ( new : unit -&gt; 'a )|The provided type must have a default constructor.|
+|Constructor Constraint|*type-parameter* : ( new : unit -&gt; 'a )|The provided type must have a parameterless constructor.|
```
Comment (cartermp) [inline docs/fsharp/language-reference/generics/constraints.md]: Please revert

File: docs/fsharp/language-reference/members/explicit-fields-the-val-keyword.md [hunk 3]
```diff
@@ -53,7 +53,7 @@ The following code shows the use of explicit fields in a class that does not hav
 
 The output is `35 22`.
 
-The following code shows the use of explicit fields in a structure. Because a structure is a value type, it automatically has a default constructor that sets the values of its fields to zero. Therefore, the `DefaultValue` attribute is not required.
+The following code shows the use of explicit fields in a structure. Because a structure is a value type, it automatically has a parameterless constructor that sets the values of its fields to zero. Therefore, the `DefaultValue` attribute is not required.
```
Comment (cartermp) [inline docs/fsharp/language-reference/members/explicit-fields-the-val-keyword.md]: please revert

File: docs/fsharp/language-reference/members/constructors.md [hunk 4]
```diff
@@ -30,7 +30,7 @@ Initialized object that has coordinates (0, 0, 0)
...
+Structures follow all the rules of classes. Therefore, you can have a primary constructor, and you can provide additional constructors by using `new`. However, there is one important difference between structures and classes: structures can have a parameterless constructor (that is, one with no arguments) even if no primary constructor is defined. The parameterless constructor initializes all the fields to the default value for that type, usually zero or its equivalent. Any constructors that you define for structures must have at least one argument so that they do not conflict with the parameterless constructor.
```
Comment (cartermp) [inline docs/fsharp/language-reference/members/constructors.md]: Please revert

**Review by cartermp — CHANGES_REQUESTED** [review]
Please revert the F# changes - F# has default constructors as a feature (unlike C# and VB), so the documentation there is accurate.

File: docs/fsharp/language-reference/attributes.md (hunk 1 above)
Comment (NRKirby) [reply docs/fsharp/language-reference/attributes.md]: I've just removed this change by amending this commit

File: docs/fsharp/language-reference/generics/constraints.md (hunk 2 above)
Comment (NRKirby) [reply docs/fsharp/language-reference/generics/constraints.md]: I've just removed this change by amending this commit

File: docs/fsharp/language-reference/members/explicit-fields-the-val-keyword.md (hunk 3 above)
Comment (NRKirby) [reply docs/fsharp/language-reference/members/explicit-fields-the-val-keyword.md]: I've just removed this change by amending this commit

File: docs/fsharp/language-reference/members/constructors.md (hunk 4 above)
Comment (NRKirby) [reply docs/fsharp/language-reference/members/constructors.md]: I've just removed this change by amending this commit

**Review by cartermp — APPROVED** [review]

File: docs/fsharp/language-reference/members/constructors.md (hunk 4 above)
Comment (rpetrusha) [reply docs/fsharp/language-reference/members/constructors.md]: There are still two remaining occurrences of "parameterless constructor" here, @cartermp and @NRKirby. Should those be reverted as well?

Comment (cartermp) [reply docs/fsharp/language-reference/members/constructors.md]: I think these might actually be accurate.

Comment (NRKirby) [reply docs/fsharp/language-reference/members/constructors.md]: @rpetrusha @cartermp I will happily revert any other changes needed.

Please can you let me know if I should actually change the other 2 occurrences please?

Comment (cartermp) [reply docs/fsharp/language-reference/members/constructors.md]: I think this is fine as-is; parameterless constructor is more accurate in the context of this kind of struct in F#.

**Review by rpetrusha — APPROVED** [review]
Thanks for contributing to the dotnet/docs repo, @NRKirby, and thanks for the responses, @cartermp. I'll merge this PR now. The changes should be live on docs.microsoft.com in the next day or two.

**Pull request merged**